+ Patch Description: Tests were failing with kernel panic in loop device.
+ So modified the tests to use root filesystem.
diff --git a/testcases/kernel/syscalls/preadv2/preadv203.c b/testcases/kernel/syscalls/preadv2/preadv203.c
index 91618f5f5..a8c468e66 100644
--- a/testcases/kernel/syscalls/preadv2/preadv203.c
+++ b/testcases/kernel/syscalls/preadv2/preadv203.c
@@ -240,6 +240,8 @@ static void check_preadv2_nowait(int fd)

 static void setup(void)
 {
+       rmdir(MNTPOINT);
+       SAFE_MKDIR(MNTPOINT, 0644);
        char path[1024];
        char buf[CHUNK_SZ];
        unsigned int i;
@@ -268,6 +270,7 @@ static void do_cleanup(void)
                if (fds[i] > 0)
                        SAFE_CLOSE(fds[i]);
        }
+       SAFE_RMDIR(MNTPOINT);
 }

 TST_DECLARE_ONCE_FN(cleanup, do_cleanup);
@@ -276,9 +279,5 @@ static struct tst_test test = {
        .setup = setup,
        .cleanup = cleanup,
        .test_all = verify_preadv2,
-       .mntpoint = MNTPOINT,
-       .mount_device = 1,
-       .all_filesystems = 1,
-       .needs_tmpdir = 1,
        .needs_root = 1,
 };

